src/pages/api/deletePagesById.ts

```diff
@@ -0,0 +1,74 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import connect from '../../server/config/db';
+import Page from '../../server/models/pages';
+import authMiddleware from '../../server/middleware/tokenVerify';
+
+
+
+// Default function for the Delete API endpoint
+export default async function deleteContent(req: NextApiRequest, res: NextApiResponse) {
+    try {
+           // Check if the request method is DELETE
+        if (req.method !== 'DELETE') {
+            return res.status(405).json({
+                message: 'Method Not Allowed',
+            });
+        }
+
+            // Connect to the database
+        await connect();
+
+        // Call the auth middleware
+        await authMiddleware(req, res);
+        
+        // pageis and section in query
+        const { pageId, sectionId } = req.query;
+
+        if (!pageId && !sectionId) {
+            return res.status(400).json({
+                message: 'Bad Request. Please provide at least pageId in the query parameters.',
+            });
+        }
+
+        if (pageId) {
+            if (sectionId) {
+                // Both pageId and sectionId provided, delete specific section
+                const updatedData = await Page.findOneAndUpdate(
+                    { '_id': pageId },
+                    { $pull: { 'sections': { _id: sectionId } } },
+                    { new: true }
```

Comment:
what does this new true is doing and delete is removing or it is softDelete?

Reply:
Page.findOneAndUpdate: - finds a single document that matches the specified query and updates.
'_id': pageId: searching for a document with the specified _id (pageId).
$pull: The update operation is { $pull: { 'sections': { _id: sectionId } } }. This uses the $pull operator to remove an element from an array.
sections: array where the _id matches the specified sectionId.
{ new: true }: option ensures that the modified document is returned after the update operation.
And it's not a soft Delete it's totally removed that perticular data from db